core/src/main/java/io/opencensus/tags/TagContext.java

```diff
@@ -47,30 +49,39 @@
   }
 
   /**
-   * Returns a builder based on this {@code TagContext}.
+   * Returns the current {@code TagContext}.
    *
-   * @return a builder based on this {@code TagContext}.
+   * @return the current {@code TagContext}.
    */
-  public Builder toBuilder() {
-    return new Builder(getTags());
+  public static TagContext getCurrentTags() {
```

Comment:
So testing is a minor thing in my opinion compared with the benefit that an indirection gives us. One extra reason to have an indirection and implement it as a no-op is that we can enable this by default in gRPC with 0 cost unless users are linking the implementation. With the current proposal users will pay the cost of tagging even if they do not use it which is in my opinion bad.